engine/jobmaster/dm/dm_jobmaster.go

```diff
@@ -119,22 +114,30 @@ func (jm *JobMaster) InitImpl(ctx context.Context) error {
 		return err
 	}
 	if err := jm.preCheck(ctx, jm.initJobCfg); err != nil {
-		return err
+		return jm.Exit(ctx, frameModel.WorkerStatus{Code: frameModel.WorkerStatusError}, err)
 	}
-	if err := jm.checkpointAgent.Create(ctx); err != nil {
+	if err := jm.taskManager.OperateTask(ctx, dmpkg.Create, jm.initJobCfg, nil); err != nil {
 		return err
 	}
-	return jm.taskManager.OperateTask(ctx, dmpkg.Create, jm.initJobCfg, nil)
+	return jm.checkpointAgent.Create(ctx)
```

Comment:
will the task failed because the checkpoint is created too later?

Reply:
revert